SERVER_EXTENSIONS.md

```diff
@@ -0,0 +1,242 @@
+      # Performs any activation that needs to happen once when the language server is booted
+      sig { override.void }
+      def activate
+      end
+
+      # Returns the name of the extension
+      sig { override.returns(String) }
+      def name
+        "Ruby LSP My Gem"
+      end
+    end
+  end
+end
+```
+
+### Enhancing features
+
+All Ruby LSP requests are listeners that handle specific node types. To enhance a request, the extension must create and
+register a listener that will collect extra results that will be automatically appended to the base language server
+response.
+
+For example: to add a message on hover saying "Hello!" on top of the base hover behavior of the Ruby LSP, we can use the
+following listener implementation.
+
+```ruby
+# frozen_string_literal: true
+
+module RubyLsp
+  module MyGem
+    # All listeners have to inherit from ::RubyLsp::Listener
+    class Hover < ::RubyLsp::Listener
+      extend T::Sig
+      extend T::Generic
+
+      ResponseType = type_member { { fixed: T.nilable(::RubyLsp::Interface::Hover) } }
+
+      # Register this listener class in the hover request
+      ::RubyLsp::Requests::Hover.add_listener(self)
+
+      sig { override.returns(ResponseType) }
+      attr_reader :response
+
+      # Listeners are initialized with the EventEmitter. This object is used by the Ruby LSP to emit the events when it
+      # finds nodes during AST analysis. Listeners must register which nodes they want to handle with the emitter (see
+      # below).
+      # Additionally, listeners are instantiated with a message_queue to push notifications (not used in this example).
+      # See "Sending notifications to the client" for more information.
+      sig { params(emitter: RubyLsp::EventEmitter, message_queue: Thread::Queue).void }
+      def initialize(emitter, message_queue)
+        super
+
+        @response = T.let(nil, ResponseType)
+
+        # Register that this listener will handle `on_const` events (i.e.: whenever a constant is found in the code)
+        emitter.register(self, :on_const)
+      end
+
+      # Listeners must define methods for each event they registered with the emitter. In this case, we have to define
+      # `on_const` to specify what this listener should do every time we find a constant
+      sig { params(node: SyntaxTree::Const).void }
+      def on_const(node)
+        # Certain helpers are made available to listeners to build LSP responses. The classes under `RubyLsp::Interface`
+        # are generally used to build responses and they match exactly what the specification requests.
+        contents = RubyLsp::Interface::MarkupContent.new(kind: "markdown", value: "Hello!")
+        @response = RubyLsp::Interface::Hover.new(range: range_from_syntax_tree_node(node), contents: contents)
+      end
+    end
+  end
+end
+```
+
+### Registering formatters
+
+Gems may also register a formatter to be used by the Ruby LSP. To do that, the extension must create a formatter runner
```

Comment:
```suggestion
Gems may also register a provide to be used by the Ruby LSP. To do that, the extension must create a formatter runner
```
(since we say `register` in the next sentence).